### ISSUE TITLE
REPL/console via hardware UART

### ISSUE THREAD

**Issue by dhalbert** [adafruit/circuitpython#4761]
Provide support for accessing the REPL over UART. There are some compile switches for this, but I don't know how complete it is on the inside. There was support long ago because of the ESP8266. The BLE-only workflow would use a similar internal path.

One request: https://github.com/adafruit/circuitpython/issues/4752#issuecomment-841111515

Related to #721.

**Comment by tannewt** [comment]
Yup, I may rethink this with the BLE work. It may involve giving UARTs a lifetime similar to displays.

**Comment by fkpwolf** [comment]
I also want this feature since virtula/emulated UART is not stable when device usb is plugin and out. I can't debug device if error happened at the time.

**Comment by tannewt** [comment]
@fkpwolf Please file a separate issue for what is unstable. It shouldn't be.

**Comment by tannewt** [comment]
We do have this at build time through DEBUG_UART. We don't currently allow boot.py to set it up though.

**Comment by HilmarSt** [comment]
After more than 5 years / 8 years (#721) , what is the status for "REPL over UART" (eventually enabled in boot.py)?

Probably MicroPython already has this with the `os.dupterm()` function, but I would like to stick to CircuitPython.

**Comment by dhalbert** [comment]
It is possible to enable this at compile-time, but not yet at run-time.

@HilmarSt What more specifically is your use case?

**Comment by HilmarSt** [comment]
My use case:
I would like to have a remote access to a RP2040 with two "HC-12" (UART to 433 MHz) modules.
The HC-12 modules offer a long-range transparent UART connection, that means I might have a REPL "everywhere".
